From bf920dd14aefad9e159f92717745465e0b03d2f6 Mon Sep 17 00:00:00 2001
From: Bartosz Golaszewski <brgl@bgdev.pl>
Date: Wed, 13 Jul 2022 16:31:42 +0200
Subject: [PATCH 2/2] v4l2: remove duplicate symbols from nvidia extensions

The jetpack headers redefine certain v4l2 symbols. Remove them when
building jepture.

Signed-off-by: Bartosz Golaszewski <brgl@bgdev.pl>
---
Upstream-Status: Inappropriate [yocto-specific]

 .../include/v4l2_nv_extensions.h              | 59 -------------------
 1 file changed, 59 deletions(-)

diff --git a/jetson_multimedia_api/usr/src/jetson_multimedia_api/include/v4l2_nv_extensions.h b/jetson_multimedia_api/usr/src/jetson_multimedia_api/include/v4l2_nv_extensions.h
index 5617508..b49bd0d 100644
--- a/jetson_multimedia_api/usr/src/jetson_multimedia_api/include/v4l2_nv_extensions.h
+++ b/jetson_multimedia_api/usr/src/jetson_multimedia_api/include/v4l2_nv_extensions.h
@@ -176,26 +176,6 @@ enum v4l2_mpeg_video_h265_profile {
 #define V4L2_H264_SPS_FLAG_FRAME_MBS_ONLY           0x10
 #define V4L2_H264_SPS_FLAG_MB_ADAPTIVE_FRAME_FIELD      0x20
 #define V4L2_H264_SPS_FLAG_DIRECT_8X8_INFERENCE         0x40
-struct v4l2_ctrl_h264_sps {
-    __u8 profile_idc;
-    __u8 constraint_set_flags;
-    __u8 level_idc;
-    __u8 seq_parameter_set_id;
-    __u8 chroma_format_idc;
-    __u8 bit_depth_luma_minus8;
-    __u8 bit_depth_chroma_minus8;
-    __u8 log2_max_frame_num_minus4;
-    __u8 pic_order_cnt_type;
-    __u8 log2_max_pic_order_cnt_lsb_minus4;
-    __s32 offset_for_non_ref_pic;
-    __s32 offset_for_top_to_bottom_field;
-    __u8 num_ref_frames_in_pic_order_cnt_cycle;
-    __s32 offset_for_ref_frame[255];
-    __u8 max_num_ref_frames;
-    __u16 pic_width_in_mbs_minus1;
-    __u16 pic_height_in_map_units_minus1;
-    __u8 flags;
-};
 
 #define V4L2_H264_PPS_FLAG_ENTROPY_CODING_MODE              0x0001
 #define V4L2_H264_PPS_FLAG_BOTTOM_FIELD_PIC_ORDER_IN_FRAME_PRESENT  0x0002
@@ -205,31 +185,6 @@ struct v4l2_ctrl_h264_sps {
 #define V4L2_H264_PPS_FLAG_REDUNDANT_PIC_CNT_PRESENT            0x0020
 #define V4L2_H264_PPS_FLAG_TRANSFORM_8X8_MODE               0x0040
 #define V4L2_H264_PPS_FLAG_PIC_SCALING_MATRIX_PRESENT           0x0080
-struct v4l2_ctrl_h264_pps {
-    __u8 pic_parameter_set_id;
-    __u8 seq_parameter_set_id;
-    __u8 num_slice_groups_minus1;
-    __u8 num_ref_idx_l0_default_active_minus1;
-    __u8 num_ref_idx_l1_default_active_minus1;
-    __u8 weighted_bipred_idc;
-    __s8 pic_init_qp_minus26;
-    __s8 pic_init_qs_minus26;
-    __s8 chroma_qp_index_offset;
-    __s8 second_chroma_qp_index_offset;
-    __u8 flags;
-};
-
-struct v4l2_ctrl_h264_scaling_matrix {
-    __u8 scaling_list_4x4[6][16];
-    __u8 scaling_list_8x8[6][64];
-};
-
-struct v4l2_h264_weight_factors {
-    __s8 luma_weight[32];
-    __s8 luma_offset[32];
-    __s8 chroma_weight[32][2];
-    __s8 chroma_offset[32][2];
-};
 
 struct v4l2_h264_pred_weight_table {
     __u8 luma_log2_weight_denom;
@@ -283,20 +238,6 @@ struct v4l2_ctrl_h264_slice_param {
     __u8 flags;
 };
 
-/** Defines whether the v4l2_h264_dpb_entry structure is used.
-If not set, this entry is unused for reference. */
-#define V4L2_H264_DPB_ENTRY_FLAG_ACTIVE     0x01
-#define V4L2_H264_DPB_ENTRY_FLAG_LONG_TERM  0x02
-struct v4l2_h264_dpb_entry {
-    __u32 buf_index; /**< v4l2_buffer index. */
-    __u16 frame_num;
-    __u16 pic_num;
-    /** @note `v4l2_buffer.field` specifies this field. */
-    __s32 top_field_order_cnt;
-    __s32 bottom_field_order_cnt;
-    __u8 flags; /* V4L2_H264_DPB_ENTRY_FLAG_* */
-};
-
 struct v4l2_ctrl_h264_decode_param {
     __u32 num_slices;
     __u8 idr_pic_flag;
-- 
2.34.1

